src/shared/analytics/google-analytics.ts

```diff
@@ -0,0 +1,185 @@
+import {
+  GOOGLE_ANALYTICS_API_SECRET,
+  GOOGLE_ANALYTICS_MEASUREMENT_ID,
+} from 'src/env/config';
+import type { Options } from 'ky';
+import ky from 'ky';
+import { version } from 'src/shared/packageVersion';
+import type { Account } from 'src/background/account/Account';
+import { Loglevel, logTable, logToConsole } from '../logger';
+import { deviceIdStore } from './shared/DeviceIdStore';
+import { omitNullParams } from './shared/omitNullParams';
+
+if (!GOOGLE_ANALYTICS_API_SECRET) {
+  // eslint-disable-next-line no-console
+  console.warn('GOOGLE_ANALYTICS_API_SECRET env var not found');
+}
+if (!GOOGLE_ANALYTICS_MEASUREMENT_ID) {
+  // eslint-disable-next-line no-console
+  console.warn('GOOGLE_ANALYTICS_MEASUREMENT_ID env var not found');
+}
+
+const USE_PAYLOAD_VALIDATION_ENDPOINT = false;
+
+type GoogleAnalyticsEvent =
+  | 'first_open'
+  | 'page_view'
+  | 'signed_message'
+  | 'signed_transaction';
+
+class GoogleAnalyticsApi {
+  baseParams?: Record<string, unknown>;
+  deviceId?: string;
```

Comment:
I think better to be explicit
```suggestion
  baseParams: null | Record<string, unknown>;
  deviceId: null | string;
```